telcell/models/simplemodel.py

```diff
@@ -0,0 +1,146 @@
+        measurement.timestamp)
+    return MeasurementPair(closest_measurement, measurement)
+
+
+def pair_measurements_based_on_time(track_a: Track,
+                                    track_b: Track) \
+        -> List[MeasurementPair]:
+    """
+    Pairs two tracks based on the time difference between the measurements.
+    It pairs all measurements from track_b to the closest pair of track_a,
+    meaning that not all measurements from track_a have to be present in the
+    final list!
+
+    @param track_a: A history of measurements for a single device.
+    @param track_b: A history of measurements for a single device.
+    @return: A list with all paired measurements.
+    """
+    paired_measurements = []
+    for measurement in track_b.measurements:
+        new_pair = make_pair_based_on_time_difference(track_a,
+                                                      measurement)
+        paired_measurements.append(new_pair)
+    return paired_measurements
+
+
+def filter_delay(paired_measurements: List[MeasurementPair],
+                 min_delay: datetime.timedelta,
+                 max_delay: datetime.timedelta) \
+        -> List[MeasurementPair]:
+    """
+    Filter the paired measurements based on a specified delay range. Can
+    return an empty list.
+
+    @param paired_measurements: A list with all paired measurements.
+    @param min_delay: the minimum amount of delay that is allowed.
+    @param max_delay: the maximum amount of delay that is allowed.
+    @return: A filtered list with all paired measurements.
+    """
+    return [x for x in paired_measurements
+            if min_delay <= x.time_difference <= max_delay]
+
+
+def measurement_pairs_with_rarest_location_per_interval_based_on_track_history(
+        paired_measurements: List[MeasurementPair],
+        interval: List[datetime.datetime],
+        history_track: Track,
+        round_lon_lats: bool) -> MeasurementPair:
+    """
+    @param paired_measurements: A list with all paired measurements to
+           consider.
+    @param interval: the interval of time for which one measurement pair must
+           be chosen.
+    @param history_track: the whole history of the right track to find rarity
+           of locations in the interval considered.
+    @param round_lon_lats: Can be toggled to round the lon/lats to two decimals
+    @return: The measurement pair that has the rarest location based on the
+             history.
+
+    TODO There is a problem with testdata, because those are almost continuous
+         lat/lon data,
+    making rarity of locations not as straightforward.
+    Pseudo-solution for now: round lon/lats to two decimals and determine
+    rarity of those.
+    This should not be used if locations are actual cell-ids
+    """
+
+    def is_timestamp_in_interval(track_timestamp, interval):
+        if interval[0] <= track_timestamp <= interval[1]:
+            return True
+
+    def is_pair_in_interval(measurement_pair, interval):
```

Comment:
Volgens deze definitie kan een pair dus in twee intervallen tegelijk zitten. Is dat gewenst?